### ISSUE TITLE
n_jobs > 2, predict issue

### ISSUE THREAD

**Issue by jianswang** [automl/auto-sklearn#655]
in automl.py
```
     predictions = self.ensemble_.predict(all_predictions)
```
in ensemble_selection.py, 
```
    def predict(self, predictions): 
        predictions = np.asarray(predictions)

        # if predictions.shape[0] == len(self.weights_),
        # predictions include those of zero-weight models.
        if predictions.shape[0] == len(self.weights_):
            return np.average(predictions, axis=0, weights=self.weights_)
```
I notice the predictions returned are not correct due to the predict_proba for each model not being accurately weighted with self.weights for n_jobs>2. Do you know the reason behind this?

**Comment by mfeurer** [comment]
Thanks for posting this. Could you please give a bit more detail (what exactly is going wrong) and maybe a mini example to reproduce this?

CC @ahn1340

**Comment by jianswang** [comment]
I think it would not be easy to notice with example. But I roughly know where is causing this issue. It should be the sorting issue for when n_jobs>2. Let's say we have 2 successful algorithms in the final ensemble with the weights respectively as below.
identifier: (17953265, 1) A 
identifier: (41231231, 1) B 
self.weights_ = [0.95(A) , 0.05(B)]
But in the calculation of predict_proba for final ensemble, in ensemble_selection.py,
```
    def get_selected_model_identifiers(self):
        output = []

        for i, weight in enumerate(self.weights_):
            identifier = self.identifiers_[i]
            if weight > 0.0:
                output.append(identifier)
        output.sort(reverse=True, key=lambda t: t[0])

        return output
```
Sorting from here causes A to be weighted as 0.05 and B to be 0.95 because the sorting swaps from [(17953265, 1),(41231231, 1)] to [(41231231, 1),(17953265, 1)].

**Comment by mfeurer** [comment]
CC @pgijsbers @ja-thomas

**Comment by mfeurer** [comment]
Fixed via #656.
